src/xtgeo/grid3d/grid_property.py

```diff
@@ -71,39 +90,98 @@ def _data_reader_factory(fformat):
+    fformat: Optional[str] = None,
+    **kwargs: Any,
+) -> GridProperty:
+    """Make a GridProperty instance directly from a file import.
+
+    Note that the the property may be linked to its geometrical grid
+    through the ``grid=`` option. Sometimes this is required, for instance
+    for most Eclipse input.
+
+    Parameters:
+        pfile:
+            Name of file to be imported.
+        fformat:
+            File format to be used (roff/init/unrst/grdecl).
+            Defaults to None and tries to infer from file extension.
+        name (str):
+            Name of property to import
+        date (int or str):
+            For restart files, date in YYYYMMDD format. Also
+            the YYYY-MM-DD form is allowed (string), and for Eclipse,
+            mnemonics like 'first', 'last' is also allowed.
+        grid (Grid, optional):
+            Grid object for checks. Optional for ROFF, required for Eclipse).
+        gridlink (bool):
+            If True, and grid is not None, a link from the grid
+            instance to the property is made. If False, no such link is made.
+            Avoiding gridlink is recommended when running statistics of multiple
+            realisations of a property.
+        fracture (bool):
+            Only applicable for DUAL POROSITY systems. If True
+            then the fracture property is read. If False then the matrix
+            property is read. Names will be appended with "M" or "F"
+        ijrange (list-like):
+            A list of 4 numbers (i1, i2, j1, j2) for a subrange
+            of cells to read. Only applicable for xtgcpprop format.
+        zerobased (bool):
+            Input if cells counts are zero- or one-based in
+            ijrange. Only applicable for xtgcpprop format.
+
+    Returns:
+        A GridProperty instance.
+
+    Examples::
 
-    For arguments, see :func:`GridProperty.from_file()`
+        import xtgeo
+        gprop = xtgeo.gridproperty_from_file("somefile.roff", fformat="roff")
 
-    Args:
-        pfile (str): Property file
-        args: See :func:`GridProperty.from_file()`
-        kwargs: See :func:`GridProperty.from_file()`.
+        # or
 
-    Example::
+        mygrid = xtgeo.grid_from_file("ECL.EGRID")
+        pressure_1 = xtgeo.gridproperty_from_file("ECL.UNRST", name="PRESSURE",
+                                                 date="first", grid=mygrid)
 
-        >>> import xtgeo
-        >>> myporo = xtgeo.gridproperty_from_file(
-        ...    reek_dir + '/reek_sim_poro.roff',
-        ...    name="PORO"
-        ... )
     """
-    return GridProperty._read_file(*args, **kwargs)
+    return GridProperty._read_file(pfile, fformat, **kwargs)
 
 
 def gridproperty_from_roxar(
-    project, gname, pname, realisation=0, faciescodes=False
-):  # pragma: no cover
+    projectname: str,
+    gridname: str,
+    propertyname: str,
+    realisation: int = 0,
+    faciescodes: bool = False,
```

Comment:
Super nice renames, makes it so much more clear what these arguments actually do.